File: src/main/java/com/example/ProyectoFinal/Repository/PrestamoRepository.java
package com.example.ProyectoFinal.Repository;
import com.example.ProyectoFinal.Model.Prestamo;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface PrestamoRepository extends JpaRepository<Prestamo, Long> {
    @Query("SELECT p FROM Prestamo p JOIN FETCH p.usuario JOIN FETCH p.ejemplar")
    List<Prestamo> findAllWithUsuarioAndEjemplar();
}

File: app/src/main/java/com/example/proyectofinal/Repository/PrestamoRepository.kt
package com.example.proyectofinal.Repository

import com.example.proyectofinal.interfaces.PrestamoRetrofitClient
import com.example.proyectofinal.model.Prestamo
import com.example.proyectofinal.model.PrestamoRequest
import com.example.proyectofinal.model.PrestamoResponse

class PrestamoRepository {

    suspend fun getPrestamos(): List<PrestamoResponse> {
        return PrestamoRetrofitClient.prestamoApi.getPrestamos()
    }

    suspend fun getPrestamoById(id: Long): Prestamo {
        return PrestamoRetrofitClient.prestamoApi.getPrestamoById(id)
    }

    suspend fun crearPrestamo(prestamo: PrestamoRequest): PrestamoResponse {
        return PrestamoRetrofitClient.prestamoApi.crearPrestamo(prestamo)
    }

    suspend fun eliminarPrestamo(id: Long) {
        PrestamoRetrofitClient.prestamoApi.eliminarPrestamo(id)
    }
}
